storage/src/adb/any/fixed/mod.rs

```diff
@@ -514,18 +541,18 @@ impl<
     pub async fn historical_proof(
         &self,
         op_count: u64,
-        start_loc: u64,
+        start_loc: Location,
         max_ops: NonZeroU64,
     ) -> Result<(Proof<H::Digest>, Vec<Operation<K, V>>), Error> {
-        let end_loc = std::cmp::min(op_count, start_loc.saturating_add(max_ops.get()));
+        let end_loc = std::cmp::min(op_count, start_loc.as_u64() + max_ops.get());
```

Comment:
Can you define end_loc as a Location here?  (It's a little odd to have start_loc a location but end_loc a u64.)